### ISSUE TITLE
Dynamic template within replace layout renderer

### ISSUE THREAD

**Issue by koryukov** [NLog/NLog#4042]
There is an unexpected behavior of [Replace Layout Renderer] when [Var Layout Renderer] is used within `searchFor` option. 

**NLog version**: 4.7.2  

**Platform**: .NET Core 3  

**Current NLog config**:

```xml
<?xml version="1.0" encoding="utf-8" ?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd" xsi:schemaLocation="NLog NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      autoReload="true"
      internalLogLevel="Off"
      internalLogFile="${basedir}/Logs/nlog.log">
  <variable name="secret" value="" />
  <targets>
    <target name="Console"
            xsi:type="Console"
            encoding="utf-8"
            layout="${pad:padding=5:inner=${uppercase:${level}}} ${replace:inner=${message}:searchFor=${var:secret}:replaceWith=****}" />
  </targets>
  <rules>
    <logger name="*" minlevel="Trace" writeTo="Console" />
  </rules>
</nlog>
```

**Code snippet to reproduce**:

```c#
public class Program
{
    private static readonly Logger logger = LogManager.GetCurrentClassLogger();

    public static void Main(string[] args)
    {
        LogManager.Configuration.Variables["secret"] = "MYSECRET"; // In real life the value is loaded from application's configuration
        logger.Info("MYSECRET should be hidden anywhere.");
    }
}
```

It displays following line:
> INFO secret:replaceWith=****}

But expected message is:
> INFO **** should be hidden.

If I set `secret` value within **nlog.config** and use `${secret}` instead of `${var:secret}` it works as expected. But I need the variable will be updated through API after apptlication's configuration binding.

[Replace Layout Renderer]: https://github.com/NLog/NLog/wiki/Replace-Layout-Renderer
[Var Layout Renderer]: https://github.com/nlog/nlog/wiki/Var-Layout-Renderer

**Comment by snakefoot** [comment]
Yes NLog has some special magic with static variables, that works all properties independent of their underlying type. The dynamic-variables only works for properties with Layout-type.

https://github.com/NLog/NLog/blob/de7f16f2bac63f7f83e360124f11abeea5e51a77/src/NLog/LayoutRenderers/Wrappers/String%20transformations/ReplaceLayoutRendererWrapper.cs#L66

Here dynamic-variable would work:

https://github.com/NLog/NLog/blob/de7f16f2bac63f7f83e360124f11abeea5e51a77/src/NLog/Targets/EventLogTarget.cs#L150

#3656 is trying to resolve #1811 and #1234

**Comment by koryukov** [comment]
@snakefoot, thank you for response. I'm ready to wait the feature implementation, becouse I'm not sure that my pull-request will be  correct.

**Comment by snakefoot** [comment]
Maybe apply the same logic, that now have been implemented for Separator with basic Layout support: #4583
